### PR TITLE
Create validations and model specs

### PR DESCRIPTION
**Pull request by Bellagirl-maker**
Add the following validations:
For the User model:
Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.
For the Post model:
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.
Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,17 +1,22 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
   has_many :comment_entries, dependent: :destroy
-  has_many :like_entries, dependent: :destroy
 
-  after_save :update_user_posts_counter
+  validates :author, presence: true
+  validates :title, presence: true
```
Comment (rotimiazeez) [inline app/models/post.rb]: Please note the title validation should include must not exceed 250 characters. 👍🏽

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Bellagirl-maker 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- `User` model has validations 👍🏽
- `Post` model has validations 💯
- unit tests implemented for `User`, `Post`, `Comment`, and `Like` model 👏🏽
- All tests passed ✔️

You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: app/models/post.rb (hunk 1 above)
Comment (Bellagirl-maker) [reply app/models/post.rb]: @rotimiazeez done with this change

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Bellagirl-maker, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Validation tests were written for necessary models. :heavy_check_mark:
-  All the models tests passed. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: _mark: 
-  Correct Gitfow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**